app/src/main/java/com/example/marcinwisniewski/intivetrainingapp/moviedetails/viewmodel/DetailsViewModel.kt

```diff
@@ -1,5 +1,24 @@
 package com.example.marcinwisniewski.intivetrainingapp.moviedetails.viewmodel
 
+import android.arch.lifecycle.MutableLiveData
 import android.arch.lifecycle.ViewModel
+import android.databinding.ObservableField
+import com.example.marcinwisniewski.intivetrainingapp.movielist.model.Movie
 
-class DetailsViewModel : ViewModel()
\ No newline at end of file
+class DetailsViewModel : ViewModel() {
+    private val choosenMovie: MutableLiveData<Movie> = MutableLiveData()
+    private var movie: ObservableField<Movie> = ObservableField()
+    fun getMovie(): Movie? {
+        return movie.get()
+    }
+
+    fun setMovie(movie: Movie) {
+        this.movie.set(movie)
+    }
+
+    fun fetchChoosenMovie(choosenMovie: Movie?): MutableLiveData<Movie> {
```

Comment:
I think this method should not return anything, since the method is called from fragment. You should store the data in viewmodel, you can't pass any data to Fragment. Fragment is a View, view can only observe the data.

Reply:
fixed